nuclear/roles/NUClearRole.cmake

```
@@ -22,33 +21,29 @@ function(NUCLEAR_ROLE)
   # Remove the :: from each module to make a valid target name for the module
   string(REPLACE "::" "" role_module_targets "${role_modules}")
 
-  # Get separate components of role path to properly set the output folder, target name and executable
-  get_filename_component(role_name ${role} NAME)
-  cmake_path(GET role PARENT_PATH role_folder)
-
-  # Target name must be unique and can't be a path, so if this is in a folder then change the slashes to underscores
-  string(REPLACE "/" "_" target_name "${role}")
-
   # Build our executable from the generated role
-  add_executable(${target_name} "${role}.cpp")
-  set_target_properties(${target_name} PROPERTIES OUTPUT_NAME ${role_name})
+  add_executable(${role} "${role}.cpp")
+  set_target_properties(${role} PROPERTIES OUTPUT_NAME ${role_name})
+  set_target_properties(${role} PROPERTIES RUNTIME_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/bin/${role_path}")
 
-  target_include_directories(${target_name} PRIVATE "${PROJECT_SOURCE_DIR}/${NUCLEAR_MODULE_DIR}")
+  # Make it possible for the role to find all of the module includes
+  target_include_directories(${role} PRIVATE "${PROJECT_SOURCE_DIR}/${NUCLEAR_MODULE_DIR}")
 
   # Link to the roles module libraries and the shared utility and extension libraries
-  target_link_libraries(${target_name} Threads::Threads)
-  target_link_libraries(${target_name} ${role_module_targets})
-  target_link_libraries(${target_name} NUClear::nuclear)
-
-  # Set our output directory to be bin
-  set_target_properties(${target_name} PROPERTIES RUNTIME_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/bin/${role_folder}")
+  target_link_libraries(${role} Threads::Threads)
+  target_link_libraries(${role} ${role_module_targets})
+  target_link_libraries(${role} NUClear::nuclear)
 
   # IDE folder
-  set_target_properties(${target_name} PROPERTIES FOLDER "roles/${role_folder}")
+  if(role_path)
+    set_property(TARGET ${role} PROPERTY FOLDER "roles/${role_path}/")
+  else()
+    set_property(TARGET ${role} PROPERTY FOLDER "roles/")
+  endif()
 
   # Store the used NUClear modules on the target as a property This can be used later in scripts to work out what
   # modules are used in the role
-  set_target_properties(${target_name} PROPERTIES NUCLEAR_MODULES "${role_modules}")
+  set_property(TARGET ${role} PROPERTY NUCLEAR_MODULES ${role_modules})
```

You can also combine this line with the other `set_target_properties` above
```suggestion
  set_target_properties(${role} PROPERTIES NUCLEAR_MODULES ${role_modules})
```